modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/DropwizardServerGenerator.scala

```diff
@@ -353,11 +355,38 @@ object DropwizardServerGenerator {
                 val responseName = s"${handlerName}.${operationId.capitalize}Response"
                 val responseType = JavaParser.parseClassOrInterfaceType(responseName)
 
+                val entitySetterIfTree = NonEmptyList
+                  .fromList(responses.value.collect({
+                    case Response(statusCodeName, Some(_)) => statusCodeName
+                  }))
+                  .map(_.reverse.foldLeft[IfStmt](null)({
```

Comment:
Before you scream at me for using `null` as a `foldLeft` initializer, let me explain.  This actually does make sense and is weirdly elegant.  The constructor signature for `IfStmt` is
```
public IfStmt(Expression condition, Statement thenStmt, Statement elseStmt)
```
So for the last item in the list, `null` has to be passed for `elseStmt`.  In the old old old similar code, I mapped the response-with-entity list into a `List[IfStmt]` and passed `null` for `elseStmt` each time, and then did a side-effect-ful `reduceLeft` to just call `prevStmt.setElse(curStmt)` to mutate the existing statements.  It worked, but I considered that ugly, as well as requiring an extra pass over the list.

This feels more elegant to me: reverse the list of stuff, pass `null` as the initializer, and then pass that as `elseStmt` in the lambda body.  One iteration over the list, and no side-effect-ful mutation later.

Reply:
I definitely agree with you, especially considering the use of `NonEmptyList`. I wish that were able to be captured in the types, but all I've got is
```scala
def elif(elseClause: IfStmt, statusCodeName: String): IfStmt = ???
...map(_.reverse match { case NonEmptyList(last, rest) => rest.foldLeft(elif(null, last))(elif _)
```
which requires another definition. The only danger of what you have here is if `NonEmptyList` is relaxed, we leak null

Reply:
Relaxed in the sense that they allow a NEL to be empty??